tsl/src/chunk.c

```diff
@@ -421,3 +422,134 @@ chunk_drop_replica(PG_FUNCTION_ARGS)
 
 	PG_RETURN_VOID();
 }
+
+typedef struct
+{
+	Chunk *chunk;
+	NameData operation_id;
+	List *src_node;
+	List *dst_node;
+} ChunkCopyData;
+
+static void
+chunk_copy_data_prepare(ChunkCopyData *ccd)
+{
+	const char *cmd;
+	const char *connection_string;
+
+	/* create publication on the source data node */
+	cmd = psprintf("CREATE PUBLICATION %s FOR TABLE %s",
+				   NameStr(ccd->operation_id),
+				   quote_qualified_identifier(NameStr(ccd->chunk->fd.schema_name),
+											  NameStr(ccd->chunk->fd.table_name)));
+	ts_dist_cmd_run_on_data_nodes(cmd, ccd->src_node, false);
+
+	/* create subscription on the destination data node */
+
+	/*
+	 * CREATE SUBSCRIPTION from a database within the same database cluster will hang,
+	 * create the replication slot separately before creating the subscription
+	 */
+	cmd = psprintf("SELECT pg_create_logical_replication_slot('%s', 'pgoutput')",
+				   NameStr(ccd->operation_id));
+	ts_dist_cmd_run_on_data_nodes(cmd, ccd->src_node, false);
```

Comment:
Just curious, what will happen on data nodes when the access node fails after executing this code?

Reply:
The replication slot needs to be created on the destination node and not on the source node. I wonder how your test script is succeeding with this..

Reply:
Previous effect of this command will still be remained and has to be manually cleaned up. We decided to address this issue separately